controllers/csm_controller.go

```diff
@@ -286,6 +286,11 @@ func (r *ContainerStorageModuleReconciler) Reconcile(_ context.Context, req ctrl
 		return utils.HandleValidationError(ctx, csm, r, err)
 	}
 
+	// Set default replica count if not specified
+	if csm.Spec.Driver.Replicas == 0 {
```

Comment:
Can we have the default value of Replicas as 2 when not set instead of setting this value explicitly

Reply:
discussions around this result in the need to do this via kubebuilder here  https://github.com/dell/csm-operator/blob/f3bcc931dff36274378f5359b8eb3bcfe510ebdb/api/v1/types.go#L188.

Reply:
@anathoodell @rishabhatdell Please test whether you are able to scale the replicas down to 0 with this change. ie deploy the driver with minimal CR, it will have 2 controller replicas. Now explicitly set the value of replicas to "0" in CSM CR and validate that the number of controllers actually scales down to 0. 
My guess is this test will fail, since the line above is setting the number of replicas to 2!

Why did we need these lines? Is the kubebuilder annotation not taking effect ?

Reply:
I checked this and you are correct the replica count is not handled properly with the latest changes.

This is Fixed in commit https://github.com/dell/csm-operator/pull/800/commits/16a6a745c75547c8645d7ed231f0aba1598d71e6.